strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ProxyRepositoryArtifactByteStreamsCopy.java

```diff
@@ -52,7 +54,9 @@
     private LayoutProviderRegistry layoutProviderRegistry;
 
     private ArtifactByteStreamsCopyStrategy simpleArtifactByteStreams = SimpleArtifactByteStreamsCopy.INSTANCE;
-
+    
+    private ThreadLocal<ArtefactCopyContext> artefactCopyContext = new ThreadLocal<>();;
```

Comment:
I have huge doubts changes in this class are related with `SB-930` :)

Reply:
yes, it's not related
I have fixed resource leaks here
also, I wanted to ask
why you implemented this with recursion, instead of just for loop?

Reply:
also, I meet this resource leak issue within this PR, so that's why it's here

Reply:
Please let me know if you find some issues with my code and I'll fix it ASAP. I'm curious where was the leak out there.

Reply:
the issue was that you using recursion there and open new connection down in the calling stack, and not close previous. I have fixed this with `ThreadLocal` context, so connection and client now shared in current thread between method calls.